### PR TITLE
update_lufo_script

### PR DESCRIPTION
**Pull request by StephanZaat**
Graag controleren op leesbaarheid.

### CODE REVIEW

File: lufopyramids.sh [hunk 1]
```diff
@@ -16,81 +23,205 @@
...
 # -----------------------------------------
+#
+#
+# When using ECW files, translate to GeoTiff manually:
+# https://hub.docker.com/r/ginetto/gdal
+# docker run --rm -it --name gdalecw -v /mnt/lufo/lufo2009/ams:/home/datafolder ginetto/gdal:2.4.4_ECW gdal_translate /home/datafolder/Amsterdam_groot.ecw /home/datafolder/lufo2009.tif
+#
+
+### Set local vars
+SOURCEDIR=$1#!/bin/bash
```
Comment (emacgillavry) [inline lufopyramids.sh:34]: Shebang is misplaced and seems a duplication of https://github.com/Amsterdam/mapserver/blob/steph/lufo_updates/lufopyramids.sh#L67.

File: lufopyramids.sh [hunk 2]
```diff
@@ -16,81 +23,205 @@
...
-# done
+# When using ECW files, translate to GeoTiff manually:
+# https://hub.docker.com/r/ginetto/gdal
+# docker run --rm -it --name gdalecw -v /mnt/lufo/lufo2009/ams:/home/datafolder ginetto/gdal:2.4.4_ECW gdal_translate /home/datafolder/Amsterdam_groot.ecw /home/datafolder/lufo2009.tif
 #
-# cp -a *.aux /mnt/lufo2020
-# cp -a *.tfw /mnt/lufo2020
 
+### Set local vars
 SOURCEDIR=$1
```
Comment (emacgillavry) [inline lufopyramids.sh:67]: See statement in review for line https://github.com/Amsterdam/mapserver/blob/steph/lufo_updates/lufopyramids.sh#L34

File: lufopyramids.sh [hunk 3]
```diff
@@ -16,81 +23,205 @@
...
-#    rm -rf $DESTDIR
-#fi
+### Put original resolution images in their own subdirectory
+mv $DESTDIR/*.tif $DESTDIR/0
 
-mkdir -p $DESTDIR/0
-gdal_retile.py -v -r cubic -levels 5 -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR --optfile /tmp/list.txt
+### Set Levels and loop over them
+### This exists to ensure public WMS usage is much quicker as the TIF files are in the blob storage
+LEVELS=$(find $DESTDIR/* -type d -exec basename {} \; | sort)
```
Comment (emacgillavry) [inline lufopyramids.sh:97]: As we only use the level 0 directory, no need for a while loop here

File: lufopyramids.sh [hunk 4]
```diff
@@ -16,81 +23,205 @@
...
-# Put original resolution images in their own subdirectory
+### Create tile indices for Mapserver WMS usage
+    gdaltindex $DESTDIR/imagery-$YEAR-$i.shp $DESTDIR/$i/*.tif
+    ogrinfo -dialect SQLITE -sql "UPDATE 'imagery-$YEAR-$i' SET location = '/vsicurl/https://files.data.amsterdam.nl/$TYPE/$YEAR/pyramid' || SUBSTR(location,length('$DESTDIR/'),length(location))" $DESTDIR/imagery-$YEAR-$i.shp
+    ogrinfo -sql "CREATE SPATIAL INDEX ON imagery-$YEAR-$i" $DESTDIR/imagery-$YEAR-$i.shp
+
+done
+
+### Create tile index for Mapserver Tiling only
+gdaltindex $DESTDIR_ROOT/imagery.shp $DESTDIR/0/*.tif
```
Comment (emacgillavry) [inline lufopyramids.sh:113]: Duplicate of https://github.com/Amsterdam/mapserver/blob/steph/lufo_updates/lufopyramids.sh#L106. Make sure to create an index file using `ogrinfo -sql "CREATE SPATIAL INDEX ON..."(sse L108) and maybe add proper location according to L107).

File: lufopyramids.sh [hunk 5]
```diff
@@ -16,81 +23,205 @@
...
+# # find . -type d -empty -delete
+# # for i in {24..25}
+# # do
+# #   for j in {15..24}
+# #   do
+# #       cp -v Ortho_1090_4917_${j}_${i}.tif ..
+# #   done
+# # done
+
+# # for f in Ortho*; do mv $f ${f/Ortho/0_Ortho}; done
```
Comment (emacgillavry) [inline lufopyramids.sh:141]: Seems you've added your worksheet below: mostly same code as above...?

File: lufopyramids.sh [hunk 6]
```diff
@@ -16,81 +23,205 @@
...
```
Comment (emacgillavry) [inline lufopyramids.sh:221]: And again the same lines of code here...?

**Review by emacgillavry — CHANGES_REQUESTED** [review]

File: lufopyramids.sh (hunk 3 above)
Comment (StephanZaat) [reply lufopyramids.sh:97]: We need the 5 levels in normal operations to provide quick WMS.
Level 0 is only used for seeding.

File: lufopyramids.sh (hunk 5 above)
Comment (StephanZaat) [reply lufopyramids.sh:141]: Right, that can be removed

**Comment by emacgillavry** [comment]
:rocket:

**Pull request merged**